CMakeLists.txt

```diff
@@ -73,6 +73,7 @@ configure_file(${VERSION_IN_FILE} ${VERSION_OUT_FILE})
 option(MIP_USE_SERIAL "Build serial connection support into the library and examples" ON)
 option(MIP_USE_TCP    "Build TCP connection support into the library and exampels" ON)
 option(MIP_USE_EXTRAS "Build extra support into the library including some things that might work at a higher level and use dynamically allocated memory" ON)
+option(MIP_ENABLE_DIAGNOSTIC_COUNTERS "Enable various diagnostic counters in the mip library." OFF)
```

Comment:
Missed this in the first review, but this does not actually enable the macro definition on the target. You still need to do something like:

```cmake
if(${MIP_ENABLE_DIAGNOSTIC_COUNTERS})
    add_compile_definitions("MIP_ENABLE_DIAGNOSTIC_COUNTERS")
endif()
```
When the option is enabled, I see some compiler errors as well

Reply:
Good catch, fixed.